src/types/events/voice_gateway/mod.rs

```diff
@@ -0,0 +1,170 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+use super::WebSocketEvent;
+use serde::{Deserialize, Serialize};
+use serde_json::{value::RawValue, Value};
+
+pub use client_connect::*;
+pub use client_disconnect::*;
+pub use hello::*;
+pub use identify::*;
+pub use media_sink_wants::*;
+pub use ready::*;
+pub use select_protocol::*;
+pub use session_description::*;
+pub use speaking::*;
+pub use ssrc_definition::*;
+pub use voice_backend_version::*;
+
+mod client_connect;
+mod client_disconnect;
+mod hello;
+mod identify;
+mod media_sink_wants;
+mod ready;
+mod select_protocol;
+mod session_description;
+mod speaking;
+mod ssrc_definition;
+mod voice_backend_version;
+
+#[derive(Debug, Default, Serialize, Clone)]
+/// The payload used for sending events to the voice gateway.
+///
+/// Similar to [VoiceGatewayReceivePayload], except we send a [Value] for d whilst we receive a [serde_json::value::RawValue]
+pub struct VoiceGatewaySendPayload {
+    #[serde(rename = "op")]
+    pub op_code: u8,
+
+    #[serde(rename = "d")]
+    pub data: Value,
+}
+
+impl WebSocketEvent for VoiceGatewaySendPayload {}
+
+#[derive(Debug, Deserialize, Clone)]
+/// The payload used for receiving events from the voice gateway.
+///
+/// Note that this is similar to the regular gateway, except we no longer have s or t
+///
+/// Similar to [VoiceGatewaySendPayload], except we send a [Value] for d whilst we receive a [serde_json::value::RawValue]
+pub struct VoiceGatewayReceivePayload<'a> {
+    #[serde(rename = "op")]
+    pub op_code: u8,
+
+    #[serde(borrow)]
+    #[serde(rename = "d")]
+    pub data: &'a RawValue,
+}
+
+impl<'a> WebSocketEvent for VoiceGatewayReceivePayload<'a> {}
+
+/// The modes of encryption available in voice udp connections;
```

Comment:
<!--
 Copyright (c) 2024 star
 
 This Source Code Form is subject to the terms of the Mozilla Public
 License, v. 2.0. If a copy of the MPL was not distributed with this
 file, You can obtain one at https://mozilla.org/MPL/2.0/.
-->


```suggestion
/// The modes of encryption available in voice UDP connections;
```